pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentTarPushJobRunner.java

```diff
@@ -84,9 +85,14 @@ public void run() {
         segmentsToPush.add(file);
       }
     }
+    Map<URI, String> uriToLineageEntryIdMap = new HashMap<>();
```

Comment:
+1, I guess this will address replication of this code. Also I have a question about this section: how do we enforce that we cant start two consistent data pushes with the same segments on the same table?

Reply:
@jugomezv The segment names should be different for different push job because we are using the timestamp as the postfix. We don't enforce one push job at the same time, and the segment replacement protocol actually handles such case. Basically, one of the push will win and ends the protocol successfully.
Example:
1. Both push create their own segments with unique names (by appending the timestamp).
2. Both job starts the protocol which `forceCleanup=true`. The `startReplaceSegments()` api guarantee atomicity based on znode versions. The first push creates an IN_PROGRESS lineage entry. The second push detects the conflict and revert the lineage entry created by the first job because we are force cleaning it up, and then create its own lineage entry. In this case the second push wins.
3. The first push job can still upload segments, but queries will not be routed to the new segments because the lineage entry is in `REVERTED` state (broker filters out the segmentsTo if the entry is reverted). It will fail finally when ending the protocol.
4.  The second push job will upload segments and end the protocol successfully.